osu.Game.Rulesets.Taiko/UI/TaikoMascotTextureAnimation.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Animations;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Skinning;
+
+namespace osu.Game.Rulesets.Taiko.UI
+{
+    public sealed class TaikoMascotTextureAnimation : TextureAnimation
+    {
+        private const float clear_animation_speed = 1000 / 10F;
+        private static readonly int[] clear_animation_sequence = { 0, 1, 2, 3, 4, 5, 6, 5, 6, 5, 4, 3, 2, 1, 0 };
+        private int currentFrame;
+
+        public TaikoMascotAnimationState State { get; }
+
+        public TaikoMascotTextureAnimation(TaikoMascotAnimationState state)
+            : base(true)
+        {
+            State = state;
+
+            // We're animating on beat if it's not the clear animation
+            if (state == TaikoMascotAnimationState.Clear)
+                DefaultFrameLength = clear_animation_speed;
+            else
+                this.Stop();
+
+            Origin = Anchor.BottomLeft;
+            Anchor = Anchor.BottomLeft;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource skin)
+        {
+            if (State == TaikoMascotAnimationState.Clear)
+            {
+                foreach (var textureIndex in clear_animation_sequence)
+                {
+                    var textureName = _getStateTextureName(textureIndex);
+                    Texture texture = skin.GetTexture(textureName);
+
+                    if (texture == null)
+                        break;
+
+                    AddFrame(texture);
+                }
+            }
+            else
+            {
+                for (int i = 0; true; i++)
+                {
+                    var textureName = _getStateTextureName(i);
+                    Texture texture = skin.GetTexture(textureName);
+
+                    if (texture == null)
+                        break;
+
+                    AddFrame(texture);
+                }
+            }
+        }
+
+        /// <summary>Advances the current frame by one.</summary>
+        public void Move()
+        {
+            if (FrameCount == 0) // Frames are apparently broken
```

Comment:
mind expounding on that comment? broken in what way? how's that comment meant to help anyone reading this source even now, let alone in a few months?

Reply:
Code would crash if the base `TextureAnimation` doesn't contain any frames, this is to prevent the code to fail and instead to just show nothing. (I encountered this while writing tests)

Reply:
The question here is whether the frame count being 0 constitutes a failure scenario (in which case it should die and it's evidence of test shortcomings) or it can sometimes happen in reasonably normal scenarios, in which case this is a sane guard and the comment can be safely removed (or replaced by a comment that explains precisely *when* an animation can have no frames).

Whether it's one or the other it's a question I or other reviewers will need to have an answer to before being able to decide with confidence to merge this code.